apps/04TowersOfHanoi.js

```diff
@@ -13,30 +13,89 @@ var stacks = {
   c: []
 };
 
+var totalMoves=0;
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack,endStack) {
   // Your code here
+  //Remove from the first stack.
+  
+  var moveChar=stacks[startStack].pop();
+  //Place on the ending stack.
+  stacks[endStack].push(moveChar)
 
+  totalMoves++;
 }
 
-function isLegal() {
+function isLegal(startStack,endStack) {
```

Comment:
Thats is a lot of code for function isLegal, a easy way to test through each parse through the stack would be the use startStack and endStack. Check for two things, is the stack empty? and is the stack i'm pushing bigger than what already exists on the stack. 
Example of how it could be done,

if(stacks[endStack].length === 0 || stacks[startStack] < stacks[endStack]){
    return true;
  }
return false;
}